Core/Business/Context.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Business;
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+class Context extends ParameterBag
+{
+    /**
+     * @var Context
+     */
+    private static $instance = null;
+
+    /**
+     * @var \Core_Foundation_IoC_Container
+     */
+    private static $container = null;
+
+    /**
+     * Get current instance of filled context (singleton)
+     *
+     * @param \Core_Foundation_IoC_Container $container The service container
+     * @return Context
+     */
+    final public static function getInstance(\Core_Foundation_IoC_Container &$container = null)
```

Comment:
Shouldn't it be the other way round?

I mean, get the context from the container instead of passing the container to the context and accessing context statically.